enhancements/api-review/gcp_user_defined_labels.md

```diff
@@ -0,0 +1,455 @@
+---
+title: gcp_user_defined_labels
+authors:
+  - "@bhb"
+reviewers:
+  - "@patrickdillon" ## reviewer for installer component
+  - "@JoelSpeed" ## reviewer for api and machine-api-provider-gcp components
+  - "@flavianmissi" ## reviewer for cluster-image-registry-operator component
+  - "@abutcher" ## reviewer for cloud-credential-operator component
+approvers:
+  - "@jerpeter1" ## approver for CFE
+api-approvers:
+  - "@JoelSpeed" ## approver for api component
+creation-date: 2022-07-12
+last-updated: 2023-04-19
+tracking-link:
+  - https://issues.redhat.com/browse/OCPPLAN-8155
+  - https://issues.redhat.com/browse/CORS-2455
+see-also:
+  - "enhancements/api-review/custom-tags-aws.md"
+  - "enhancements/api-review/azure_user_defined_tags.md"
+replaces:
+  - N/A
+superseded-by:
+  - N/A
+---
+
+# Apply user defined labels to all GCP resources created by OpenShift
+
+## Summary
+
+This enhancement describes the proposal to allow an administrator of OpenShift to
+have the ability to apply user defined labels and tags to those resources created
+by OpenShift in GCP.
+
+## Motivation
+
+Motivations include but are not limited to:
+
+- Allow admin, compliance, and security teams to keep track of assets and objects
+  created by OpenShift in GCP.
+
+### User Stories
+
+- As an openshift administrator, I want to have labels added to all resources created
+  in GCP by Openshift, so that I can query resources by the labels or filter resources
+  by label in cloud billing.
+- As an openshift administrator, I want to have tags added to all resources created
+  in GCP by Openshift, so that I can use it for creating IAM policies or to configure 
+  organization policy constraints or to refine resource usage and cost data.
+
+### Goals
+
+- The administrator or service (in the case of Managed OpenShift) installing OpenShift
+  can configure a list of up to 32 user-defined labels in the Openshift installer generated
+  install config, which is referenced and applied by installer and the in-cluster operators
+  on the the GCP resources during creation.
+- Labels must be applied at creation time, in an atomic operation. It isn't acceptable
+  to create an object and to apply labels post cluster creation.
+- The administrator or service (in the case of Managed OpenShift) installing OpenShift
```

Comment:
Should we consider breaking out tags vs labels into two different enhancements? They aren't set to achieve the same goal so should they be grouped together?

Reply:
GCP provides tags and labels which is just tags having both functionalities in AWS, Azure. And since we were updating metadata of a resource(GCP tags aren't actually resource metadata but could be viewed like one) and getting reviewed too will be helpful, so thought to have single proposal for both. Should I break it into two proposals?

Reply:
I am fine leaving them together if others are. They do go together although they are used for different purposes.